No need to call [caml_initialize] in [caml_make_vect], since we make sure the initial value is not in the minor heap

jhjourdan opened the pull request:
On the following micro-benchmark, this gives a x2.2 performance improvement on my machine:

```
let bench () =
  let t1 = Sys.time () in
  for i = 0 to 5000000 do
    ignore (Array.make 300 1)
  done;
  Sys.time () -. t1

let () =
  for i = 0 to 10 do
    let t1 = bench () in
    Printf.printf "%f\n%!" t1;
  done
```

Moreover, I would argue the code of `caml_make_vect` becomes simpler, since it somewhat factorizes the case where `init` is young or not.

jhjourdan committed:
No need to call [caml_initialize] in [caml_make_vect], since we make sure the initial value is not in the minor heap.

gasche reviewed (approved):
This seems correct to me, and I agree that the code is slightly easier to read. (I would maybe add a `CAMLassert(!(Is_young(init))` at the end of the forced-minor-gc block, but that comes from the existing code.)

jhjourdan commented:
> This seems correct to me, and I agree that the code is slightly easier to read. (I would maybe add a CAMLassert(!(Is_young(init)) at the end of the forced-minor-gc block, but that comes from the existing code.)

I added the assertion, and rebased the PR.

gasche commented:
This should be good to go once CI passes.

stedolan reviewed (approved):
This is a nice improvement!

(There's a typo in Changes: `Jourda`)

gasche commented:
I'll merge now, fix the typo and add Stephen as a reviewer directly in trunk.

gasche commented:
( bcf176cfd585a9064e64019d2003b55de4b77ba3 )

jhjourdan commented:
Thanks!
